libraries/version/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(
+  VERSION 3.5 )
+
+project(
+  Version )
+
+# Found in directory `eos/CMakeModules/`.
+include(
+  VersionUtils )
+
+# Generate the most up-to-date version metadata of the repository.
+GENERATE_VERSION_METADATA()
```

Comment:
It appears this generation is being done at cmake configure/generation and not when building. If we're going to report dirtiness we need more accuracy. To get this, will need to replicate how appbase creates a dummy target that is run on each build to populate the build information.

Reply:
Updated.